plugin.js

```diff
@@ -106,18 +115,24 @@ function plugin (fastify, options, next) {
   }
 
   fastify.decorateRequest('cookies', null)
-  fastify.decorateReply('cookie', setCookie)
+  fastify.decorateReply(kReplySetCookies, null)
 
+  fastify.decorateReply('cookie', setCookie)
   fastify.decorateReply('setCookie', setCookie)
   fastify.decorateReply('clearCookie', clearCookie)
 
   if (hook) {
     fastify.addHook(hook, onReqHandlerWrapper(fastify, hook))
+    fastify.addHook('onSend', onSendHandlerWrapper(fastify))
   }
 
   next()
 
   // ***************************
+  function serializeCookie (name, value, options) {
+    return cookie.serialize(name, value, options)
+  }
+
```

Comment:
maybe has perf improvement if we dont wrap it again in a function

Maybe this is more performant as it does not wrap it in a fn.

```suggestion
  const serializeCookie = cookie.serialize

```